nav2_behavior_tree/plugins/condition/would_a_smoother_recovery_help_condition.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2023 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License. Reserved.
+
+#include "nav2_behavior_tree/plugins/condition/would_a_smoother_recovery_help_condition.hpp"
+#include <memory>
+
+namespace nav2_behavior_tree
+{
+
+WouldASmootherRecoveryHelp::WouldASmootherRecoveryHelp(
+  const std::string & condition_name,
+  const BT::NodeConfiguration & conf)
+: AreErrorCodesPresent(condition_name, conf)
+{
+  error_codes_to_check_ = {
+    ActionGoal::UNKNOWN,
+    ActionGoal::TIMEOUT,
+    ActionGoal::FAILED_TO_SMOOTH_PATH
```

Comment:
SMOOTHED_PATH_IN_COLLISION -- if a recovery is to generate a new plan, it could help